hydra/virtual-tree.go

```diff
@@ -340,3 +368,25 @@ func (r *directoryTreeBuilder) dir(dir Directory, isRoot bool) error { //nolint:
 
 	return nil
 }
+
+// Combine creates a path from the parent combined with the relative path. The relative
+// path is a file system path so should only contain forward slashes, not the standard
+// file path separator as denoted by filepath.Separator, typically used when interacting
+// with the local file system. Do not use trailing "/".
+func Combine(parent, relative string) string {
+	if relative == "" {
+		return parent
+	}
+
+	return parent + "/" + relative
+}
```

Comment:
REJECTED. This is deliberate because these paths are used with file system apis, where the path can only contain '/', must not start or end with a '/'. Using filepath.Join is not appropriate in the case and would be wrong